File: 1319-unique-number-of-occurrences/unique-number-of-occurrences.ts
function uniqueOccurrences(arr: number[]): boolean {
    const numberMap: Record<number, number> = {};

    arr.forEach((n) => {
        numberMap[n] = (numberMap[n] ?? 0) + 1;
    });

    const uniqueMap: Record<number, boolean> = {};
    for (const [key, value] of Object.entries(numberMap)) {
        if (uniqueMap[value]) {
            return false
        }

        uniqueMap[value] = true
    }

    return true;
};

File: 1319-unique-number-of-occurrences/unique-number-of-occurrences.kt
class Solution {
    fun uniqueOccurrences(arr: IntArray): Boolean {
        val map = hashMapOf<Int, Int>()
        arr.forEach {
            map[it] = (map[it] ?: 0) + 1
        }

        val set = hashSetOf<Int>()
        map.values.forEach {
            if (set.contains(it)) {
                return false
            } else {
                set.add(it)
            }
        }
        
        return true
    }
}